spec/factories/krikri_activities.rb

```diff
@@ -1,18 +1,65 @@
 FactoryGirl.define do
   factory :krikri_activity, class: Krikri::Activity do
+    id 1
     agent 'Krikri::Harvesters::OAIHarvester'
     opts '{"uri": "http://example.org/endpoint"}'
   end
 
   factory :krikri_harvest_activity, parent: :krikri_activity do
+    id 2
     agent 'Krikri::Harvesters::OAIHarvester'
     opts '{"uri": "http://example.org/endpoint", ' \
     '"oai": {"metadata_prefix": "mods", "set": "set1"}}'
   end
 
   factory :krikri_mapping_activity, parent: :krikri_activity do
+    id 3
     agent 'Krikri::Mapper::Agent'
-    opts({ name: 'test_map', generator_uri: 'http://example.org/agent/1' }
-      .to_json)
+    opts(
+      {
+        name: 'test_map',
+        generator_uri: (RDF::URI(Krikri::Settings['marmotta']['ldp']) /
+          Krikri::Settings['prov']['activity'] / '1').to_s
+      }.to_json
+    )
   end
+
+  # FIXME:
+  #
+  # The following `factory' will cause a FactoryGirl::InvalidFactoryError error
+  # because "Krikri::Enrichments::StripHtml" fails the validation of the
+  # Activity model, because it's not the name of a class that extends or
+  # includes Krikri::SoftwareAgent.
+  # (See Krikri::Activity#agent_must_be_a_software_agent)
+  # 
+  # e.g.:
+  #
+  # [4] pry(main)> Krikri::Mapper::Agent < Krikri::SoftwareAgent          
+  # => true
+  # [2] pry(main)> Krikri::Enrichments::StripHtml < Krikri::SoftwareAgent  
+  # => nil
+  # [7] pry(main)> Krikri::Enrichment < Krikri::SoftwareAgent
+  # => nil
+  # [9] pry(main)> Krikri::Mapper::Agent.include?(Krikri::SoftwareAgent)
+  # => true
+  # [10] pry(main)> Krikri::Enrichments::StripHtml.include?(Krikri::SoftwareAgent)
+  # => false
+  # [11] pry(main)> e = Krikri::Enrichments::StripHtml.new
+  # => #<Krikri::Enrichments::StripHtml:0x007fa08ef3c100>
+  # [12] pry(main)> e.respond_to?(:agent_name)
+  # => false
+  #
+  # It's not clear from the inline enrichment comments how to enqueue
```

Comment:
I could have made this more clear, but I am not suggesting changing this in this branch.

I am suggesting that it gets addressed somewhere else so that I can rebase upon it.  The fact is we have enrichment classes that appear to be software agents, that can not be represented in this factory, and it just makes sense to have them accounted for here if we're already doing harvests and mappings.

Following the example of the existing `mapper_agent_spec.rb`, which was written to verify that a mapper agent can get the records generated by a harvest activity, it only made sense to eventually write an enrichment agent that gets the records generated by a mapping activity, and an indexing agent that gets the records generated by an enrichment activity.